jupyter_server/serverapp.py

```diff
@@ -1578,9 +1554,40 @@ def init_shutdown_no_activity(self):
                           self.shutdown_no_activity_timeout)
             pc = ioloop.PeriodicCallback(self.shutdown_no_activity, 60000)
             pc.start()
+    
+    def init_httpserver(self):
+        """initialize tornado httpserver"""
```

Comment:
I would add a note here that `init_webapp` *must* be called before this in order to ensure that multiple config/setting variables get initialized and set up. It should also specify what the alternative is (e.g. if you pass `new_httpserver=False`, then how is `self.http_server` supposed to be set?).